### ISSUE TITLE
OpenAIService does not gracefully close when parent thread / Transient dep injected service closes

### ISSUE THREAD

**Issue by dannyr-git** [betalgo/openai#173]
**Describe the bug**
Exceptions are thrown when a transient service exists that leverages OpenAIService for a group of calls and the thread goes unused and naturally closes.

**Your code piece**
```
        private readonly IOpenAIService? _openAIService;
        private readonly ILocalSettings settingsService;
        private readonly ILoggingService Logger;
        private string _apikey;
        private bool _disposed;

        public OpenAIAPIService(ILocalSettings settingsService, ILoggingService logger)
        {
            this.settingsService = settingsService;
            this.Logger = logger;
            _apikey = settingsService.Load<string>("ApiKey");

            if (String.IsNullOrEmpty(_apikey))
            {
                _apikey = "Api Key Is Null or Empty";
                Logger.LogError("_apikey");
            }

            _openAIService = new OpenAIService(new OpenAiOptions()
            {
                ApiKey =  _apikey
            });
        }
```
**Result**
A series of socket exceptions occur :
(The only socket code in this app is OpenAIService

![image](https://user-images.githubusercontent.com/15082948/225512073-ec4f4d4a-66be-4c60-b9a0-838aebcec6c6.png)

**Expected behavior**
OpenAIService needs to gracefully close or be IDisposable 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [Windows]
 - Language [C#]
 - Version [6.8.0]

**Additional context**

**Comment by kayhantolga** [comment]
https://github.com/betalgo/openai/pull/243

**Comment by kayhantolga** [comment]
fixed in v7.0.0 https://github.com/betalgo/openai/pull/271
